### ISSUE TITLE
library scanning is restricted to -depth 1

### ISSUE THREAD

**Issue by jgilmore** [andrewrk/groovebasin#492]
My library structure is mostly collections of CD's.  for instance:
~/Music/Weird Al Yankovic/Dare To Be Stupid/09 This Is The Life.mp3
~/Music/Audiobooks/Brandon_Mull-Beyonders-A_World_Without_Heroes/Track 001-cd 01.mp3
~/Music/Musicals/Evita-Premiere American Release/2-01-On The Balcony Of The Casa Rosada-Don't Cry For Me Argentina.mp3

NONE of these show up in the player. If I move the album directories ("Dare To Be Stupid") to ~/Music then it gets added and will show up for a search. That's a pretty major restructuring that I'd like to avoid.

**Comment by thejoshwolfe** [comment]
This is already supposed to work the way you want. Is there any chance there are symlinks involved?

**Comment by thejoshwolfe** [comment]
Also check the permissions of the directories. particularly the 'x' permission.

**Comment by andrewrk** [comment]
Symlinks are supposed to work fine too.

**Comment by jgilmore** [comment]
OK. Got after screwing around with the latest, I've reached he following conclusion:

Library scanning takes a really long time. And there's no way to tell when it's done.

If I added the recursive flag to the watcher: 
diff --git a/lib/player.js b/lib/player.js
index 4e23237..6eb63c2 100644
--- a/lib/player.js
+++ b/lib/player.js
@@ -778,7 +780,7 @@ Player.prototype.watchDirEntry = function(dirEntry) {
   var fullDirPath = path.join(self.musicDirectory, dirEntry.dirName);
   var watcher;
   try {
-    watcher = fs.watch(fullDirPath, onChange);
+    watcher = fs.watch(fullDirPath, {recursive: true}, onChange);
     watcher.on('error', onWatchError);
   } catch (err) {
     log.warn("Unable to fs.watch:", err.stack)


Then it did take nearly as long? Might be purely subjective, I suppose. In any case, this is either a non-issue, or the incorrect issue. (The correct one being "there's no way to tell if library scanning is done, other than the player being responsive")
